db/migrate/20170927135007_migrate_dialog_field_associations_to_use_new_relationship.rb

```diff
@@ -0,0 +1,31 @@
+class MigrateDialogFieldAssociationsToUseNewRelationship < ActiveRecord::Migration[5.0]
+  Rails.application.eager_load!
+
+  def up
+    Dialog.all.each do |dialog|
+      fields = dialog.dialog_fields
+      dialog_fields_with_associations = fields.select { |df| !df.auto_refresh.nil? || !df.trigger_auto_refresh.nil? }.sort { |n| n.position * -1 }
+      triggers = dialog_fields_with_associations.select { |df| df.trigger_auto_refresh == true }
+      responders = dialog_fields_with_associations.select { |df| df.auto_refresh == true }
+      triggers.each_with_index do |t, index|
+        responder_fields_for_t = if !triggers[index + 1].nil?
```

Comment:
Prefer coding in the positive as opposed to `! nil?` because double-negative are hard for people to reason about.  So,

```ruby
responder_fields_for_t =
  if triggers[index + 1]
    responders.select { |r| r.position
 ....
```